app/pubspec.yaml

```diff
@@ -8,7 +8,10 @@
 
 name: sharezone
 description: Der digitale Schulplaner &quot;Sharezone&quot; für Android und iOS.
-version: 1.6.2+316
+# Note that this build number is not the latest build number. Please go to App
+# Store Connect or use "app-store-connect get-latest-build-number 1434868489" to
+# get the latest build number.
+version: 1.6.2+318
```

Comment:
Yeah, I also don't like the current situation. My plan add a `sz release` command (see #387) which takes care of all the release naming, build number, etc. process so that you don't need to care about this anymore. However, this a bit more complex

Reply:
But this would have the same problem wouldn't it? (That the build number in the code is not correct)   

I guess there is no 100% safe way to always have them in sync (since you can always manually upload a new version somewhere). Maybe a little crazy idea: Couldn't we have a bot that automatically increments the build number on the main branch? Or is this stupid?

Of course we could also let the `sz release` command change the actual file as well so that the developer will hopefully commit this change later.

Idk

Reply:
> But this would have the same problem wouldn't it? (That the build number in the code is not correct)

Yes, but I think in the long term it should be possible to publish a new version via our CD (requires #387) which result into that you don't need the current build number in the code (it's useless - what's the point to have the current build number in the code -> would be only helpful to manually release the version (without `sz release`) and even in this rare case you could look up the latest build number).

I thought a long time about this and all solutions have with disadvantages ._. When you would like to have the current build number in the code, you would need to update the code after publishing the version. Either you push a commit directly to the `main` branch which would require to allow to push directly to `main` (not good) or you open a PR every time. And then you need to be aware of not coming into an endless loop of updates because you updated the `pubspec.yaml` -> need to detect if only the build number changed, if yes, don't make an update. And think about the cases where you have multiple releases in the pipeline that this doesn't make any race conditions.

Especially, when we have a `sz release` command, having the current latest build number in the code has no advantage (only if you don't want to use `sz release` and don't want to use our CD and for only this case you would need to deal with a bunch of disadvantages (named already a few) and only to not look up the build number).